src/librustc_back/target/arm_unknown_linux_musleabi.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use target::Target;
+
+pub fn target() -> Target {
+    let mut base = super::linux_base::opts();
+
+    // These settings are copied from the x86_64_unknown_linux_musl target.
+    base.pre_link_args.push("-nostdlib".to_string());
+    base.pre_link_args.push("-static".to_string());
+
+    base.pre_link_args.push("-Wl,--eh-frame-hdr".to_string());
+
+    base.pre_link_args.push("-Wl,-(".to_string());
+    base.post_link_args.push("-Wl,-)".to_string());
+
+    base.pre_link_objects_exe.push("crt1.o".to_string());
+    base.pre_link_objects_exe.push("crti.o".to_string());
+    base.post_link_objects.push("crtn.o".to_string());
+    base.dynamic_linking = false;
+    base.has_rpath = false;
+    base.position_independent_executables = false;
+
+    // Most of these settings are copied from the arm_unknown_linux_gnueabi
+    // target.
```

Comment:
If you're feeling ambitious, perhaps this could become a standard among our musl targets? They could take the corresponding Linux target and then modify the same set of options to become "musl specific"

It's somewhat unfortunate to duplicate both the definition of the target and the musl configuration in multiple places :(. Not your fault or anything of course, so can always wait for a future PR.